tests/provisioning/provision_test.py

```diff
@@ -0,0 +1,119 @@
+class TestGKECluster(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        """ Get the location of GKE cluster for later queries """
+        command = ('gcloud container clusters list --format="value(location)"')
+        result = subprocess.run(split(command), encoding='utf-8', capture_output=True)
+        cls.location = str(result.stdout.replace('\n', ''))
+        if cls.location == '':
+            raise Exception('GKE cluster is not provisioned!')
+        cls.name = 'projects/' + getProjectId() + '/locations/' + cls.location + '/clusters/stackdriver-sandbox'
+    
+    def testNodeMachineType(self):
+        """ Test if the machine type for the nodes is as specified """
+        client = cluster_manager.ClusterManagerClient()
+        cluster = client.get_cluster(name=self.__class__.name)
+        node_config = cluster.node_config
+        self.assertEqual(node_config.machine_type, 'n1-standard-2')
+
+    def testNumberOfNode(self):
+        """ Test if the number of nodes in the node pool is in the specified range """
+        client = cluster_manager.ClusterManagerClient()
+        cluster = client.get_cluster(name=self.__class__.name)
+        self.assertTrue(cluster.current_node_count == 4)
+    
+    def testStatusOfServices(self):
+        """ Test if all the service deployments are ready """
+        command = ("kubectl get deployment --all-namespaces -o json")
+        result = subprocess.run(split(command), encoding='utf-8', capture_output=True)
+        services = json.loads(result.stdout)
+        for service in services['items']:
+            self.assertTrue(int(service['status']['readyReplicas']) >= 1)
+    
+    def testReachOfFrontend(self):
+        """ Test if query frontend returns 200 """
+        command_ip = ("kubectl get svc frontend-external -o json")
+        result_ip = subprocess.run(split(command_ip), encoding='utf-8', capture_output=True)
+        info = json.loads(result_ip.stdout)
+        external_ip = info['status']['loadBalancer']['ingress'][0]['ip']
+        port = info['spec']['ports'][0]['port']
+        url = "".join(["http://", external_ip, ':', str(port)])
+        self.assertTrue(urllib.request.urlopen(url).getcode() == 200)
+
+class TestLoadGenerator(unittest.TestCase):
+    def testReachOfLoadgen(self):
+        """ Test if query load generator returns 200 """
+        command_ip = ('gcloud compute instances list --filter="name:loadgenerator*" --format="value(networkInterfaces[0].accessConfigs.natIP)"')
+        result_ip = subprocess.run(split(command_ip), encoding='utf-8', capture_output=True)
+        url = "".join(['http://', result_ip.stdout.split('\n')[0], ':8080'])
+        self.assertTrue(urllib.request.urlopen(url).getcode() == 200)
+
+class TestProjectResources(unittest.TestCase):
+    def testCloudTrace(self):
+        """ Test if Cloud Trace is provisioned """
+        try:
+            command = ('gcloud alpha trace sinks list --project={0}'.format(getProjectId()))
```

Comment:
What does this command return? Is it returning the traces that were captured in a project?
Also, if this is not available in gcloud beta or gcloud (ga), make sure you do
```
gcloud components alpha install
```
before the tests start